### PR TITLE
feat: add configuration for EPHEMERAL volume

### LINKED ISSUE #9261: implement configuration for EPHEMERAL volume
First round of configuration options available.

The configuration is only respected when the EPHEMERAL volume is missing completely.

Configurable options:

* disk selector expression
* minimum size
* maximum size
* automatic growth

(encryption is already handled)

### PR DESCRIPTION
**Pull request by smira**
Fixes #9261

More info in the docs.

### CODE REVIEW

File: pkg/machinery/config/types/block/volume_config.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+		Meta: meta.Meta{
+			MetaKind:       VolumeConfigKind,
+			MetaAPIVersion: "v1alpha1",
+		},
+	}
+}
+
+func exampleVolumeConfigEphemeralV1Alpha1() *VolumeConfigV1Alpha1 {
+	cfg := NewVolumeConfigV1Alpha1()
+	cfg.MetaName = "remote-log"
```
[the remark this answers is not in the document]
Comment (smira) [reply pkg/machinery/config/types/block/volume_config.go:123]: hmm... I have no idea how it got there :exploding_head: should have been EPHEMERAL

File: internal/integration/api/volumes.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func (suite *VolumesSuite) testDiscoveredVolumes(node string) {
 			Names: []string{"xfs"},
 		},
 		"EPHEMERAL": {
-			Names: []string{"xfs"},
+			Names: []string{"xfs", ""},
```
Comment (frezbo) [inline internal/integration/api/volumes.go:74]: :sweat_smile:

**Review by frezbo — APPROVED** [review]

**Review by Unix4ever — APPROVED** [review]

File: internal/integration/api/volumes.go (hunk 2 above)
Comment (smira) [reply internal/integration/api/volumes.go:74]: I gave up. it doesn't affect anything in real life except for the test

Comment (frezbo) [reply internal/integration/api/volumes.go:74]: yeh, hopefully we find a good reproducer in future

**Pull request merged**